src/JustSaying/IAmJustSaying.cs

```diff
@@ -29,5 +31,9 @@ public interface IAmJustSaying : IMessagePublisher
         IMessageLockAsync MessageLock { get; set; }
 
         IMessageContextAccessor MessageContextAccessor { get; set; }
+
+        void SetMessageBackoffStrategy(IMessageBackoffStrategy value);
+
+        void SetOnError(Action<Exception, SQSMessage> value);
```

Comment:
AWS types have always been avoided on public interfaces in JustSaying, can this be changed to just Message?

Reply:
This is a weird one, I'm not sure why it's not included somewhere like IMessageMonitor instead. It comes from [here](https://github.com/gkinsman/JustSaying/blob/channel-message-pump/src/JustSaying/AwsTools/QueueCreation/SqsReadConfiguration.cs#L26) and is passed to the MessageDispatcher.
Looks like we call it if the message can't be serialised so can't use `Message`, but without knowing how people might be using this it's hard to know what to do with it. Maybe adding another method to IMessageMonitor with the region/queue or even the message Body?

Reply:
Yeah, it looks like `OnError` directly exposes the AWS `Message` to consumers. Reviewing most of the usages of it at the moment, and they look to be logging an error. This should definitely be moved to `IMessageMonitor` if it is the case that its all just logging.